packages/core/src/indexing/local-index-api.ts

```diff
@@ -136,34 +154,33 @@ export class LocalIndexApi implements IndexApi {
     }
   }
 
-  indexedModels(): Array<StreamID> {
+  indexedModels(): Array<ModelFieldsIndex> {
     return this.databaseIndexApi?.getIndexedModels() || []
   }
 
-  async indexModels(models?: Array<StreamID>): Promise<void> {
-    if (!models) {
-      return
-    }
-
+  async indexModels(models: Array<ModelFieldsIndex>): Promise<void> {
     const modelsNoLongerIndexed = await this.databaseIndexApi?.getModelsNoLongerIndexed()
 
-    const indexModelsArgs = []
-    for (const modelStreamId of models) {
-      this.logger.imp(`Starting indexing for Model ${modelStreamId.toString()}`)
-      const indexModelArgs = await _getIndexModelArgs(modelStreamId, this.repository)
-      if (modelsNoLongerIndexed) {
-        const modelNoLongerIndexed = modelsNoLongerIndexed.some(function (streamId) {
-          return streamId.equals(modelStreamId)
-        })
-        // TODO(CDB-2297): Handle a model's historical sync after re-indexing
-        if (modelNoLongerIndexed) {
-          throw new Error(
-            `Cannot re-index model ${modelStreamId.toString()}, data may not be up-to-date`
-          )
+    const indexModelsArgs = await Promise.all(
+      models.map(async (idx) => {
+        const modelStreamId = idx.streamID
```

Comment:
this is getting kind of hard to read - can we make a `_indexModel` helper function to clean this up a bit?